app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -241,6 +242,27 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         });
     }
 
+    // Return true is system wide dark theme is enabled else false
+    private boolean getSystemDefaultThemeBool(String theme) {
```

Comment:
These methods are repeated across classes, making future updates to this functionality more difficult.

I would extract a class `NightModeChecker` and have it be injected

Reply:
@macgills I tried to avoid the repetition but **getResources()** says a non-static method can't be inside a static method!

Reply:
Then don't use static. Create a normal class to host the logic, one that is instantiated by Dagger

Reply:
Done!